crates/routes/src/images.rs

```diff
@@ -225,9 +226,22 @@ pub async fn image_proxy(
   // for arbitrary purposes.
   RemoteImage::validate(&mut context.pool(), url.clone().into()).await?;
 
+  if context
+    .settings()
+    .pictrs_config()?
+    .proxy_bypass_domains
+    .iter()
+    .any(|s| url.domain().expect("Invalid URL").starts_with(s))
```

Comment:
Panicing for non-domain hosts is unnecessary

```suggestion
    .any(|s| url.domain().is_some_and(|domain| domain.starts_with(s)))
```